spaceship/views.py

```diff
@@ -353,10 +353,13 @@ def register():
 
     if not email_confirmed:
       token = generate_confirmation_token(email_address)
-      email.send(to_emails=email_address,
-          subject='Please verify you email for Spaceship Earth',
-          html_content=render_template('confirm_email.html',
-          confirmation_url=url_for('confirm_email', token=token, _external=True)))
+      EmailJob.enqueue(
```

Comment:
(After factoring out business logic I think we may end up in a state like "emails.send_account_verification(...)", so prolly the view code won't really know anything about jobs.)

Reply:
:+1: to factoring the business logic out of the views. i am still in favor of broader classes for flows, rather than individual functions. for instance, there's an account onboarding flow which sends an account verification email, and *it's the only thing which sends those emails*. so there's no good reason to expose a `send_account_verification` method in a grab-bag module called `emails`. also, there's no reason to have to open a random `emails` file when you're working on the account onboarding flow.

i would be in favor of, like, a class that's like `AccountOnboarding.perform(...)` and the `send_account_verification` function lives in that class along with other functions that have to do with account onboarding.

Reply:
Hmm yeah account_onboarding may make more sense. 🙂 But then there’s still views.py, which is horizontal not vertical... and that has a certain charm because views are all peer level concerns that connect a lot. anyways happy to do what works best. 😊

also I would be remiss if I didn’t link this great yegge rant here, lol:
http://steve-yegge.blogspot.com/2006/03/execution-in-kingdom-of-nouns.html?m=1